File: backend/IELTS.AI.Evaluator.Data/Models/WritingPrompt.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace IELTS.AI.Evaluator.Data.Models
{
    public class WritingPrompt : BaseEntity
    {
        public string Topic { get; set; } = default!;
        public string QuestionType { get; set; } = default!;
        public string QuestionText { get; set; } = default!;
        public string TaskType { get; set; } = "Task2"; // Or "Task1"
        public string Level { get; set; } = "Academic"; // or "General"
        public string? ImageUrl { get; set; } // For Task 1 prompts that require an image
    }

}

File: frontend/ielts-ai-evaluator-frontend/src/types/WritingPrompt.ts
// Types matching backend WritingPromptDto / WritingPromptUpsertRequest (WritingPromptService.cs)
export default interface WritingPrompt {
  writingPromptId: string;
  topic: string;
  description: string;
  preview: string;
  questionType: string;
  questionText: string;
  duration: number;
  minimumWords: number;
  taskType: "Task1" | "Task2";
  level: "Academic" | "General";
  imageUrl?: string;
  imageDescription?: string;
  isActive: boolean;
  createdAt: string;
  updatedAt: string;
}

// Body for POST /api/writing-prompts (admin upsert)
export interface WritingPromptUpsertRequest {
  writingPromptId?: string;
  topic: string;
  description: string;
  preview: string;
  questionType: string;
  questionText: string;
  duration: number;
  minimumWords: number;
  taskType: "Task1" | "Task2";
  level: "Academic" | "General";
  imageUrl?: string;
  imageDescription?: string;
  isActive: boolean;
}
